tests/ballerina-tools-integration-test/src/test/java/org/ballerinalang/test/packaging/PackagingInitTestCase.java

```diff
@@ -0,0 +1,249 @@
+
+    @BeforeClass()
+    public void setUp() throws BallerinaTestException, IOException {
+        serverZipPath = System.getProperty(Constant.SYSTEM_PROP_SERVER_ZIP);
+        tempProjectDirectory = Files.createTempDirectory("bal-test-integration-packaging-project-");
+        // Set values to manifest
+        manifest = new Manifest();
+        manifest.setName("integrationtests");
+        manifest.setVersion("0.0.0");
+    }
+
+    @Test(description = "Test creating a project with a main in a package")
+    public void testInitWithMainInPackage() throws Exception {
+        Path projectPath = tempProjectDirectory.resolve("firstTestWithPackagesMain");
+        Files.createDirectories(projectPath);
+
+        sourceFiles = new ArrayList<>();
+        packageDescFiles = new ArrayList<>();
+
+        sourceFiles.add(new SrcFile("foo", FileType.MAIN));
+        packageDescFiles.add(new PackageMdFile("foo", FileType.MAIN));
+
+        InitHandler.initialize(projectPath, manifest, sourceFiles, packageDescFiles);
+
+        Assert.assertTrue(Files.exists(projectPath.resolve("foo").resolve("main.bal")));
+        Assert.assertTrue(Files.exists(projectPath.resolve("Ballerina.toml")));
+        Assert.assertTrue(Files.exists(projectPath.resolve("foo").resolve("tests").resolve("main_test.bal")));
+
+        ballerinaServerForMain = new ServerInstance(serverZipPath);
+        String[] clientArgs = {"--sourceroot", projectPath.toString(), "foo"};
+        logLeecher = new LogLeecher("Hello World!");
+        ballerinaServerForMain.addLogLeecher(logLeecher);
+        ballerinaServerForMain.runMain(clientArgs, getEnvVariables(), "run");
+    }
+
+    @Test(description = "Test creating a project with a service in a package")
+    public void testInitWithServiceInPackage() throws Exception {
+        Path projectPath = tempProjectDirectory.resolve("firstTestWithPackagesService");
+        Files.createDirectories(projectPath);
+
+        sourceFiles = new ArrayList<>();
+        packageDescFiles = new ArrayList<>();
+
+        sourceFiles.add(new SrcFile("foo", FileType.SERVICE));
+        packageDescFiles.add(new PackageMdFile("foo", FileType.SERVICE));
+
+        InitHandler.initialize(projectPath, manifest, sourceFiles, packageDescFiles);
+
+        Assert.assertTrue(Files.exists(projectPath.resolve("foo").resolve("hello_service.bal")));
+        Assert.assertTrue(Files.exists(projectPath.resolve("Ballerina.toml")));
+        Assert.assertTrue(Files.exists(projectPath.resolve("foo").resolve("tests").resolve("hello_service_test.bal")));
+
+        ballerinaServerForService = ServerInstance.initBallerinaServer();
+        ballerinaServerForService.startBallerinaServer(projectPath.resolve("foo").resolve("hello_service.bal")
+                                                                  .toString());
+        HttpResponse response = HttpClientRequest.doGet(ballerinaServerForService.getServiceURLHttp
+                ("hello/sayHello"));
```

Comment:
This is not the correct way to break new lines. We usually break it with period (.).